dashboard/app/controllers/api_controller.rb

```diff
@@ -432,24 +432,29 @@ def user_progress_for_lesson
 
       # Temporarily return the full set of progress so we can overwrite what the sessionStorage changed
       response[:progress] = summarize_user_progress(script, current_user)[:progress]
+      response[:isHoc] = script.hoc?
 
       if user_level
         response[:lastAttempt] = {
           timestamp: user_level.updated_at.to_datetime.to_milliseconds,
           source: level_source
         }
-      end
-      response[:isHoc] = script.hoc?
 
-      recent_driver, recent_attempt, recent_user = UserLevel.most_recent_driver(script, level, current_user)
-      if recent_driver
-        response[:pairingDriver] = recent_driver
-        if recent_attempt
-          response[:pairingAttempt] = edit_level_source_path(recent_attempt)
-        elsif level.channel_backed?
-          @level = level
-          recent_channel = get_channel_for(level, script.id, recent_user) if recent_user
-          response[:pairingChannelId] = recent_channel if recent_channel
+        # Pairing info
+        is_navigator = user_level.navigator?
+        if is_navigator
+          driver = user_level.driver
+          driver_level_source_id = user_level.driver_level_source_id
+        end
+
+        response[:isNavigator] = is_navigator
+        if driver
```

Comment:
Is `driver` still set if the current user is the driver?

Reply:
No, because driver is set above on in line 446 inside the `is_navigator`.  I'm open to other ways of structuring this if this is confusing?

Reply:
If the current user is the driver, then we don't set `pairingDriver`?

Reply:
If the current user is the driver, we do not want to set `pairingDriver` in the response to the client. 

This used to be more important because the client used the existence of `pairingDriver` to determine if the current user was the navigator.  There is now an explicit flag that indicates if the current user is a navigator so it would be ok to always send `pairingDriver` but I left the previous behavior that `pairingDriver`, `pairingAttempt`, and `pairingChannelId` are only sent if the current user is the navigator.